src/credentials/FabricTable.h

```diff
@@ -220,18 +220,69 @@ class DLL_EXPORT FabricInfo
         return TLV::EstimateStructOverhead(sizeof(uint16_t), Crypto::P256SerializedKeypair::Capacity());
     }
 
-    NodeId mNodeId           = kUndefinedNodeId;
-    FabricId mFabricId       = kUndefinedFabricId;
-    FabricIndex mFabricIndex = kUndefinedFabricIndex;
+    static void StaticAsserts()
+    {
+        static_assert(sizeof(void *) == 4 || sizeof(void *) == 8,
+                      "Unexpected pointer width, computation of expected size will be off.");
+
+        constexpr size_t expected_size = []() -> size_t {
+            // mNodeId, mFabricId, mCompressedFabricId are 8 bytes each.
+            size_t totalSize = 8 * 3;
+
+            // Root public key is a vtable pointer followed by 65 bytes of data,
+            // but the space it takes up needs to end on a void * alignment
+            // boundary.  So it's 72 (= 69 + 3) bytes on 32-bit and 80 (= 73 + 7)
+            // bytes on 64-bit.
+            if constexpr (sizeof(void *) == 4)
+            {
+                totalSize += 72;
+            }
+            else
+            {
+                totalSize += 80;
+            }
+
+            // mFabricLabel is 33 bytes and has no alignment requirements.
+            totalSize += 33;
+
+            // mFabricIndex, mVendorId, mHasExternallyOwnedOperationalKey are
+            // 1, 2, 1 bytes respectively, and doing them in that order has no
+            // alignment issues.
+            totalSize += 1 + 2 + 1;
+
+            // mOperationalKey has to be aligned on a void * alignment
```

Comment:
This is all ABI specific. Just curious why the expected size checks? The reordering is safe without all of this. Why is this present at all?

Reply:
The idea was to (1) verify that the size is in fact shrinking and (2) ensure that people don't accidentally regress it (esp. by adding one boolean member and making the size 8 bytes larger or something like that).

Concretely, there are some booleans I would like to add to FabricInfo, and knowing that I am not actually changing its size in the process is useful.

But I could remove this too and just test locally.

Reply:
OK, simplified to one very simple assert about the expected padding existing, and added some comments explaining the layout instead.